pkg/instancetype/instancetype.go

```diff
@@ -227,43 +227,27 @@ func (m *InstancetypeMethods) storePreferenceRevision(vm *virtv1.VirtualMachine)
 }
 
 func GenerateRevisionNamePatch(instancetypeRevision, preferenceRevision *appsv1.ControllerRevision) ([]byte, error) {
-	var patches []patch.PatchOperation
+	var opts []patch.PatchOption
```

Comment:
Another options is to create the obj with a base (even an empty one) and then add to it options before generating the payload:

```go
patchSet := patch.New(...)
if foo {
    patchSet.AddOption(WithTest(...))
}
payload := patchSet.GeneratePayload()
```

Reply:
Done